include/stellar/utils/fraction.hpp

```diff
@@ -76,6 +85,17 @@ struct fraction
         return from_double(value);
     }
 
+    static fraction from_double_with_limit(long double value, uint32_t const precision_limit)
+    {
+        // Applying a lower bound to the error rate because values close to 0 increase runtime dramatically  
+        if (abs(value) < (1.0 / (precision_limit * 1.05)))
```

Comment:
`abs` is C and just for ints.
```suggestion
        if (std::abs(value) < (1.0 / (precision_limit * 1.05)))
```